STYLE_GUIDE.md

```diff
@@ -0,0 +1,218 @@
+
+Because Kedro-Viz can be imported into other applications, we avoid ever setting global styles. We use the `pipeline-` prefix for most HTML/CSS classes, to reduce the risk of clashes in the global namespace when this project is imported into other applications. It's important never to use global element selectors or generic classes like `.header`.
+
+In general, we try to avoid splitting up blocks and elements in our class names, in order make the full class-name easier to search. However in general, splitting modifiers is okay.
+
+Avoid:
+
+```scss
+.pipeline-nodelist {
+  &__list {
+    @extend %nolist;
+
+    &--nested {
+      margin: 0 0 1.2em;
+    }
+  }
+}
+```
+
+Prefer:
+
+```scss
+.pipeline-nodelist__list {
+  @extend %nolist;
+
+  &--nested {
+    margin: 0 0 1.2em;
+  }
+}
+```
+
+### Typography
+
+Typography styles are set using the `.kedro` class. It should wrap all text elements, in order to enforce a consistent font-size/font-family etc, allowing the font-sizes to be set relatively using `em` but keeping them independent of parent app font styles. This class is [inherited from Kedro-UI](https://github.com/quantumblacklabs/kedro-ui/blob/master/src/styles/typography/styles.css), where it's used to establish a base of 10px allowing all other `em` based units to equal the equivalent in pixels divided by ten.
+
+### Units
+
+In general, we try to use `em` for styles that depend on font-size (e.g. text margins), `%` for relative layout positioning, and `px` for absolute layout positioning. Avoid using `rem` or viewport units (e.g. `vh`/`vw`/`vmin`/`vmax`) in code that's included in the package library, as these can cause side-effects if the parent app modifies the base font-size or the app is not full-screen.
+
+### Variables
+
+We use Sass variables to declare colours and reuse values wherever useful. It's not necessary to declare every CSS unit value as a variable, but if a value is used in multiple places (for related reasons), then it makes sense to make it a variable - either locally in the component, or application-wide in `src/styles/_variables.scss`. Using a named variable instead of a number clarifies intent, and helps avoid updates to the value in multiple places later on.
+
+Avoid:
+
+```scss
+color: #001521;
+color: rgba(#001521, 0.5);
+color: rgba(0, 0, 0, 0.5);
+```
+
+Prefer:
+
+```scss
+color: $color-bg-dark-1;
+color: rgba($color-bg-dark-1, 0.5);
+color: rgba(black, 0.5);
+```
+
+### Theme colours
+
+Kedro-Viz has both light and dark themes, and every component must be styled appropriately for each theme. The themes are set using the same theme classes that Kedro-UI uses - `kui-theme--light` and `kui-theme--dark` - applied on the top-level container element. To simplify code and avoid increasing CSS specificity, we use CSS custom properties to handle theme colours. We recommend that you use these too where possible - they are defined in the App component.
```

Comment:
I think it might be a nice idea to leave a blank line after this first sentence to stress the importance of this.